docs/git-branches-conventions.md

```diff
@@ -0,0 +1,59 @@
+# Git branch conventions
+
+There are 3 types of branches:
+
+1. The master branch
+2. Hotfix branches
+3. Development branches (to be merged into master and then deleted)
+
+If your contribution is a hotfix (bug fix on a release already published), follow the
+[hotfix branches](#hotfix-branches) section.
+In all other cases, you must create a development branch, and submit a pull request to
+master (possibly in "draft" state until you are done).
+
+## Hotfix branches
+
+To fix a bug in a release already published (in github releases page),
+the way to proceed will depend on whether this release has already be subject to hotfix or not:
+
+### First hotfix
+
+If you need to make a hotfix on a release that never received one,
+you must first create a hotfix branch dedicated to that release
+(from the commit pointed by the tag of the concerned release.), this branch must be named
+`hotfix-vX.Y`, where `X` and `Y` must be replaced by the major and minor version numbers
+respectively.
+
+Then you need to create a branch named `{name}-hotfix-vX.Y{-optional-words}`, where `{name}` is to
+be replaced by your nickname, and then create a PR from this branch to the hotfix branch.
+This procedure allows to play the CI to check the integrity of a hotfix.
+
+Example:
+
+You have to make a first hotfix on the release v2.3.0 and your nickname is kevin. We suppose that
+it is for example a bug in the `key` subcommand.
+
+1. Create a branch name `hotfix-v2.3`
+2. Create a branch name `kevin-hotfix-v2.3-key-command`
+3. Create a PR from `kevin-hotfix-v2.3-key-command` into `hotfix-v2.3`.
+4. Request an emergency one or more reviews of this PR.
+5. Merge this PR.
+6. Create a hotfix tag `v2.3.1` on the `hotfix-v2.3` branch.
+7. If relevant, create an issue to backport this hotfix into master.
```

Comment:
So `hotfix-v2.3` never gets merged back into master?

What is the procedure for backporting the hotfix? I can imagine two things that both seem fine to me:
1. Make another PR from `kevin-hotfix-v2.3-key-command` into `master` and treat it like any development PR?
2. Make a brand new PR that branches off of a more recent `master` and reapply the changes as you see fit.

Reply:
> What is the procedure for backporting the hotfix?

It is not defined indeed. I think it depends on the case, sometimes it will not be relevant to backport the hotfix because in the meantime changes on master make the bug no longer relevant.
Other times the bug will still exist on master but will require a different fix, that's why I'm not in favor of reusing the same branch.
I think it's better to create a new branch (and a new ticket) to handle the backport completely independently :)